src/ast_dumper.cpp

```diff
@@ -37,7 +37,15 @@ void AstDumper::Visit(const DeclNode& decl) {
     decl.init->Accept(*this);
     indenter_.DecreaseLevel();
   }
-  std::cout << ')' << std::endl;
+  std::cout << indenter_.Indent() << ')' << std::endl;
+}
+
+void AstDumper::Visit(const LoopInitNode& loop_init) {
+  if (std::holds_alternative<std::unique_ptr<DeclNode>>(loop_init.clause)) {
+    std::get<std::unique_ptr<DeclNode>>(loop_init.clause)->Accept(*this);
+  } else {
+    std::get<std::unique_ptr<ExprNode>>(loop_init.clause)->Accept(*this);
+  }
```

Comment:
Just to let you know, the following alternative achieves the same semantics:
```cpp
  std::visit([this](const auto& node) { node->Accept(*this); },
             loop_init.clause);
```
It appears to sacrifice readability for me, so I wouldn't recommend it.